compiler/rustc_codegen_llvm/src/intrinsic.rs

```diff
@@ -212,6 +212,25 @@ impl<'ll, 'tcx> IntrinsicCallMethods<'tcx> for Builder<'_, 'll, 'tcx> {
                     ],
                 )
             }
+
+            sym::ptr_auth_asia | sym::ptr_auth_asib | sym::ptr_auth_asda | sym::ptr_auth_asdb => {
+                if !self.sess().target.options.features.contains("+pauth") {
```

Comment:
Using `contains` here is not robust and may introduce unexpected bug (https://github.com/rust-lang/rust/issues/114153).

Reply:
Could you please clarify your point?
What is the best way to check that features exist?
On that PR, they were looking for the too small pattern, weren't they?

Reply:
We can probably do the same thing that #114332 did for the `c` target feature. (i.e., [add symbol to compiler/rustc_span/src/symbol.rs](https://github.com/rust-lang/rust/pull/114332/files#diff-97beaf86f77cdca7e6b5583aa0764c8e0a493b957f6e5c1bb7ba40e15f2ed2d7R448) and then [check to see if `sess.unstable_target_features` contains them](https://github.com/rust-lang/rust/pull/114332/files#diff-6bbf6edd930f26d3b180eb5c250b0e48d8f3c5eb474e3274909ef6ae6f0d1e61R279).)

Reply:
Thank you!